source/fundamentals/crud/write-operations/insert.txt

```diff
@@ -4,9 +4,194 @@ Insert a Document
+before the error-producing document still get inserted into your collection.
+
+.. note::
+
+   You can get an acknowledgement of successful document insertion even
+   if a WriteError occurs:
+
+   .. code-block:: go
+      
+      docs := []interface{}{
+	      bson.D{{"_id", 1}, {"country", "Tanzania"}},
+	      bson.D{{"_id", 2}, {"country", "Lithuania"}},
+	      bson.D{{"_id", 1}, {"country", "Vietnam"}},
+	      bson.D{{"_id", 3}, {"country", "Argentina"}},
+	  }
+
+      result, err := coll.InsertMany(context.TODO(), docs)
+      list_ids := result.InsertedIDs
+      if err != nil {
+	      fmt.Printf("A bulk write error occurred, but %v documents were still inserted.\n", len(list_ids))
+      }
+
+      for _, id := range list_ids {
+	      fmt.Printf("Inserted document with _id: %v\n", id)
+      }
+   
+   The output should look like this:
+
+   .. code-block:: none
+      :copyable: false
+      
+      A bulk write error occurred, but 2 documents were still inserted.
+      Inserted document with _id: 1
+      Inserted document with _id: 2
+   
+   If you look inside your collection, you should be able to see the following documents:
+   
+   .. code-block:: json
+      :copyable: false
+
+      { "_id": 1, "country": "Tanzania" }
+      { "_id": 2, "country": "Lithuania" }
+
+Example
+~~~~~~~
+
+The following example creates and inserts some documents into the
+``favorite_books`` collection using the ``InsertMany()`` method:
+
+.. code-block:: go
+
+   coll := client.Database("myDB").Collection("favorite_books")
+   docs := []interface{}{
+       bson.D{{"title", "My Brilliant Friend"}, {"author", "Elena Ferrante"}, {"year_published", 2012}},
+       bson.D{{"title", "Lucy"}, {"author", "Jamaica Kincaid"}, {"year_published", 2002}},
+       bson.D{{"title", "Cat's Cradle"}, {"author", "Kurt Vonnegut Jr."}, {"year_published", 1998}},
+   }
+
+   result, err := coll.InsertMany(context.TODO(), docs)
+   list_ids := result.InsertedIDs
+   fmt.Printf("Documents inserted: %v\n", len(list_ids))
+
+   for _, id := range list_ids {
+	   fmt.Printf("Inserted document with _id: %v\n", id)
+   }
+
+Your output should look like this:
+
+.. code-block:: none
+   :copyable: false
+
+   Documents inserted: 3
+   Inserted document with _id: ObjectID("6112d6a0acf85446f904eb91")
+   Inserted document with _id: ObjectID("6112d6a0acf85446f904eb92")
+   Inserted document with _id: ObjectID("6112d6a0acf85446f904eb93")
+
+For more information on the types and functions mentioned in this
+section, see the following resources: 
+
+- `InsertMany() <https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo@v1.7.0#Collection.InsertMany>`__ API documentation
+- `InsertManyResult <https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo@v1.7.0#InsertManyResult>`__ API documentation
+- Server Manual Entry on :manual:`insertMany() </reference/method/db.collection.insertMany/>`
+- Runnable :doc:`Insert Multiple Documents Example </usage-examples/insertMany>`
```

Comment:
I'm wondering if we should continue doing this as we start go-docs -- lets ask the team tomorrow and adjust accordingly for this and the change a document page